cf_xarray/accessor.py

```diff
@@ -116,15 +116,15 @@
 
 #: regular expressions for guess_coord_axis
 regex = {
-    "time": "time[0-9]*|min|hour|day|week|month|year",
+    "time": "\\bt\\b|time[0-9]*|min|hour|day|week|month|year",
     "vertical": (
-        "(lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|isobaric|pres|"
-        "isotherm)[a-z_]*[0-9]*"
+        "\\bz\\b|(nav_)?(lev|gdep|lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|"
+        "isobaric|pres|isotherm)[a-z_]*[0-9]*"
     ),
     "Y": "y",
-    "latitude": "y?lat[a-z0-9]*",
+    "latitude": "(y|nav_)?(lat|gphi)[a-z0-9]*",
```

Comment:
```suggestion
    "latitude": "y?(nav_|lat|gphi)[a-z0-9]*",
```

?

Reply:
With this `"y?(nav_|lat|gphi)[a-z0-9]*"` I think nav_lon/lat/lev would conflict. Maybe let's keep it simple and use `"y?(nav_lat|lat|gphi)[a-z0-9]*"`